### PR TITLE
[ARITH] Improve vector simplification for float operands

### PR DESCRIPTION
**Pull request by merrymercy**

### CODE REVIEW

File: src/arith/pattern_match.h [hunk 1]
```diff
@@ -145,6 +145,14 @@ class PEqualChecker<IntImm> {
   bool operator()(const IntImm& lhs, const IntImm& rhs) const { return lhs->value == rhs->value; }
 };
 
+template <>
+class PEqualChecker<FloatImm> {
+ public:
+  bool operator()(const FloatImm& lhs, const FloatImm& rhs) const {
+    return BaseValueEqual()(lhs->value, rhs->value);
```
Comment (merrymercy) [inline src/arith/pattern_match.h:152]:
```suggestion
    return BaseValueEqual()(rhs->value, lhs->value);
```

File: src/arith/pattern_match.h [hunk 2]
```diff
@@ -145,6 +145,14 @@ class PEqualChecker<IntImm> {
   bool operator()(const IntImm& lhs, const IntImm& rhs) const { return lhs->value == rhs->value; }
 };
 
+template <>
+class PEqualChecker<FloatImm> {
+ public:
+  bool operator()(const FloatImm& lhs, const FloatImm& rhs) const {
+    return BaseValueEqual()(rhs->value, lhs->value);
```
Comment (merrymercy) [inline src/arith/pattern_match.h:152]:
```suggestion
    return BaseValueEqual()(lhs->value, rhs->value);
```

File: src/arith/pattern_match.h (hunk 1 above)
Comment (merrymercy) [inline src/arith/pattern_match.h:152]:
```suggestion
    return BaseValueEqual()(rhs->value, lhs->value);
```

File: src/arith/rewrite_simplify.cc [hunk 3]
```diff
@@ -133,6 +135,7 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const AddNode* op) {
     TVM_TRY_REWRITE(ramp(b1, s1, lanes) + broadcast(x, lanes), ramp(b1 + x, s1, lanes));
     TVM_TRY_REWRITE(broadcast(x, lanes) + ramp(b1, s1, lanes), ramp(x + b1, s1, lanes));
     TVM_TRY_REWRITE(broadcast(x, lanes) + broadcast(y, lanes), broadcast(x + y, lanes));
+    TVM_TRY_REWRITE_IF(x + broadcast(c4, lanes), x, std::fabs(c4.Eval()->value) < 1e-20);
```
Comment (tqchen) [inline src/arith/rewrite_simplify.cc:138]: can we simply use equal instead if we just want to check 0?

File: src/arith/pattern_match.h [hunk 4]
```diff
@@ -145,6 +146,14 @@ class PEqualChecker<IntImm> {
   bool operator()(const IntImm& lhs, const IntImm& rhs) const { return lhs->value == rhs->value; }
 };
 
+template <>
+class PEqualChecker<FloatImm> {
+ public:
+  bool operator()(const FloatImm& lhs, const FloatImm& rhs) const {
+    return std::fabs(rhs->value - lhs->value) < 1e-20;
```
Comment (merrymercy) [inline src/arith/pattern_match.h:153]:
```suggestion
    return std::fabs(lhs->value - rhs->value) < 1e-20;
```

**Pull request merged**